Initial commit with modified Swagger UI and updated README.md

Kevsy opened the pull request:
### PR Category
Enhancement

### What is this PR and why do we need it?
This enables CAMARA API OAS YAML files to be rendered in the 'human friendly' Swagger UI format , at a location under CAMARA control (camera.github.io). Any files which is not a CAMARA API OAS YAML will be rejected.

This is an intial PR to deliver the functionality, future PRs can modify certain UI aspects (e.g. add some CAMARA branding)

### Alternative solutions
Continue to rely on 3rd party hosting and availability for Swagger UI rendering, which is currently unreliable.

### what issue does this fix
Fixes https://github.com/camaraproject/tooling/issues/8

Kevsy committed:
Intial commit with modified Swagger UI and updated README.md

Kevsy commented:
@hdamker @MarkusKuemmerle to test, use kevsy.github.io which has the same code already deployed:

- [with CAMARA YAML](https://kevsy.github.io/?url=https://raw.githubusercontent.com/camaraproject/SimpleEdgeDiscovery/refs/heads/main/code/API_definitions/simple-edge-discovery.yaml) (success case)
- [with non-CAMARA YAML](https://kevsy.github.io/?url=https://raw.githubusercontent.com/RandyMoore/OAS/refs/heads/master/openapi/oas.yaml) (forbidden case)

hdamker committed:
Update swagger-initializer.js

Handling more error cases, e.g. empty and malformed URLs

hdamker commented:
> @hdamker @MarkusKuemmerle to test, use kevsy.github.io which has the same code already deployed:
> 
> * [with CAMARA YAML](https://kevsy.github.io/?url=https://raw.githubusercontent.com/camaraproject/SimpleEdgeDiscovery/refs/heads/main/code/API_definitions/simple-edge-discovery.yaml) (success case)
> * [with non-CAMARA YAML](https://kevsy.github.io/?url=https://raw.githubusercontent.com/RandyMoore/OAS/refs/heads/master/openapi/oas.yaml) (forbidden case)

@Kevsy thanks a lot for the PR! I have two points:

**Handling of error cases:**

* While your validation handles the two cases above, it fails (silently, with a blank page) for non, empty, and malformed URLs, so more or less in all error cases
* I've "clauded" a new version and provided it in https://github.com/Kevsy/camaraproject.github.io/pull/1 which you can merge into your branch. It covers the two cases from above and several other cases:
  * [with CAMARA YAML](https://hdamker.github.io/?url=https://raw.githubusercontent.com/camaraproject/SimpleEdgeDiscovery/refs/heads/main/code/API_definitions/simple-edge-discovery.yaml) (success case)
  * [with non-CAMARA YAML](https://hdamker.github.io/?url=https://raw.githubusercontent.com/RandyMoore/OAS/refs/heads/master/openapi/oas.yaml) (forbidden case)
  * [use with valid URL, but http](https://hdamker.github.io/?url=http://raw.githubusercontent.com/camaraproject/SimpleEdgeDiscovery/refs/heads/main/code/API_definitions/simple-edge-discovery.yaml)
  * [malformed URL](https://hdamker.github.io/?url=not-a-url)
  * [empty URL](https://hdamker.github.io/?url=not-a-url)
  * [no URL](https://hdamker.github.io/)
* BTW: after the Swagger UI is initialized with a valid URL it can be (manually) used with other URLs. As it runs completely within the browser client and the Swagger UI code is doing a proper validation we don't need to care.

**Move into a sub directory:**

* To keep the option to use the camaraproject.github.io in future also for other purposes it would be good to move the Swagger UI to a sub directory, so that the URL will be `https://camaraproject.github.io/swagger-ui/?url=...`
* If we don't do it now we would have a lot of effort to change the links across CAMARA
* Please move specific parts for the Swagger UI deployment out of the root README

Kevsy committed:
Merge pull request #1 from hdamker/main

Update swagger-initializer.js

Kevsy committed:
Moved to sub-directory and improved error handling

Kevsy commented:
Thanks @hdamker (and Claude 😁  )  - good point about using a sub-directory.  I've merged the error handling code, moved the Swagger code into `/swagger-ui` and changed the root README (which will also be the default page returned when visiting https://camara.github.io ):


# camaraproject.github.io

Repository for CAMARA GitHub Pages.

* [/swagger-ui](./swagger-ui) contains code to display CAMARA API definitions in human-readable form. 


For information about the CAMARA project please visit our
* [Website (camaraproject.org)](https://camaraproject.org)
* [Wiki (lf-camaraproject.atlassian.net)](https://lf-camaraproject.atlassian.net/)
* [GitHub (github.com/camaraproject)](https://github.com/camaraproject)

hdamker reviewed (approved):
@Kevsy LGTM ... I propose to merge and to test in one repository (I volunteer with QualityOnDemand) the needed changes. 

Regarding the change across all repositories I'm thinking about creating the necessary PRs automatically.

hdamker commented:
@Kevsy One additional point: could you put into the swagger-ui/README which exact version you have forked (I suppose a released one), so that we know when we need to update?

Kevsy committed:
Update to Swagger UI r5.22.0

Kevsy commented:
> @Kevsy One additional point: could you put into the swagger-ui/README which exact version you have forked (I suppose a released one), so that we know when we need to update?

Sure, just done (see /swagger-ui/README.md) . 

Note they made another release  just after I forked 🤦‍♂️... so I have now also updated the relevant files to the latest r5.22. Fortunately `swagger-intializer.js` was unchanged, so our modifications did not need any further modification :).

I've deployed this updated 5.22 version to [kevsy.github.io](https://kevsy.github.io) and tested, seems fine.

hdamker commented:
@Kevsy Thanks a lot! Will merge here so that I can test my workflow against a single repository (QualityOnDemand).

I have one operation which updates README.md, README.MD, CHANGELOG.MD, and CHANGELOG.md and another operation which goes through all release descriptions of a repository and replaces the links there as well.
